### PR TITLE
Adds scheduler tests

### PR DESCRIPTION
**Pull request by paullewis**

### CODE REVIEW

File: src/scheduler.js [hunk 1]
```diff
@@ -86,19 +86,25 @@ function phaseRunner(gatherers) {
   };
 }
 
+// This isn't exposed functionality, and it's not core functionality, so skipped. */
+/* istanbul ignore next */
 function saveArtifacts(artifacts) {
```
Comment (samccone) [inline src/scheduler.js]: if we are not using the code we should delete it with git it is very easy to bring back

Comment (paullewis) [inline src/scheduler.js]: We use it, it's just not exposed externally to the module.

**Comment by paullewis** [comment]
FWIW I have opened a separate issue for things we're ignoring.

**Comment by samccone** [comment]
sgtm :+1:

File: src/scheduler.js (hunk 1 above)
Comment (samccone) [inline src/scheduler.js]: ah gotcha yeaaaa fun stuff

**Pull request merged**